contracts/eden/tests/test-eden.cpp

```diff
@@ -220,6 +257,70 @@ TEST_CASE("genesis")
+{
+   eden_tester t;
+   t.genesis();
+
+   t.alice.act<actions::inductinit>(4, "alice"_n, "bertie"_n, std::vector{"pip"_n, "egeon"_n});
+   t.bertie.act<token::actions::transfer>("bertie"_n, "eden.gm"_n, s2a("10.0000 EOS"), "memo");
+   CHECK(get_eden_membership("bertie"_n).status() == eden::member_status::pending_membership);
+
+   // cannot endorse before video and profile are set
+   auto blank_hash_data =
+       eosio::convert_to_bin(std::tuple(std::string{}, eden::new_member_profile{}));
+   expect(t.alice.trace<actions::inductendorse>(
+              "alice"_n, 4, eosio::sha256(blank_hash_data.data(), blank_hash_data.size())),
+          "not set");
+
+   expect(t.bertie.trace<actions::inductinit>(5, "bertie"_n, "ahab"_n,
+                                              std::vector{"pip"_n, "egeon"_n}),
+          "inactive member bertie");
+
+   std::string bertie_video = "QmTYqoPYf7DiVebTnvwwFdTgsYXg2RnuPrt8uddjfW2kHS";
+   t.bertie.act<actions::inductprofil>(4, bertie_profile);
+   t.alice.act<actions::inductvideo>("alice"_n, 4, bertie_video);  // Can be inviter or witness
+
+   expect(t.bertie.trace<actions::inductdonate>("bertie"_n, 4, s2a("10.0000 EOS")),
+          "induction is not fully endorsed");
+
+   auto hash_data = eosio::convert_to_bin(std::tuple(bertie_video, bertie_profile));
+   auto induction_hash = eosio::sha256(hash_data.data(), hash_data.size());
+   expect(t.bertie.trace<actions::inductendorse>("alice"_n, 4, induction_hash),
+          "missing authority of alice");
+   expect(t.bertie.trace<actions::inductendorse>("bertie"_n, 4, induction_hash),
+          "Induction can only be endorsed by inviter or a witness");
+   expect(t.alice.trace<actions::inductendorse>(
+              "alice"_n, 4, eosio::sha256(hash_data.data(), hash_data.size() - 1)),
+          "Outdated endorsement");
+
+   auto endorse_all = [&] {
+      t.alice.act<actions::inductendorse>("alice"_n, 4, induction_hash);
+      t.pip.act<actions::inductendorse>("pip"_n, 4, induction_hash);
+      t.egeon.act<actions::inductendorse>("egeon"_n, 4, induction_hash);
+   };
+   endorse_all();
+   t.chain.start_block();
+   expect(t.alice.trace<actions::inductendorse>("alice"_n, 4, induction_hash), "Already endorsed");
+
+   // changing the profile resets endorsements
+   t.bertie.act<actions::inductprofil>(4, bertie_profile);
+   expect(t.bertie.trace<actions::inductdonate>("bertie"_n, 4, s2a("10.0000 EOS")),
+          "induction is not fully endorsed");
+   t.chain.start_block();
+   endorse_all();
+
+   // changing the video resets endorsements
+   t.pip.act<actions::inductvideo>("pip"_n, 4, bertie_video);
+   expect(t.bertie.trace<actions::inductdonate>("bertie"_n, 4, s2a("10.0000 EOS")),
+          "induction is not fully endorsed");
+   t.chain.start_block();
+   endorse_all();
+
+   t.bertie.act<actions::inductdonate>("bertie"_n, 4, s2a("10.0000 EOS"));
+   CHECK(get_eden_membership("bertie"_n).status() == eden::member_status::active_member);
+}
```

Comment:
awesome tests! Can you add a few more? (it does not need to be this pr)
- trying to induct an existing member
- trying to induct with the inviter being repeated in the endorsement list
- trying to induct with repeated endorsers
- trying to induct with a pending member on the endorsement list
- trying to induct with a pending member as inviter
- complete an induction where the invitee has another X inductions and make sure the data was gc properly, when X has the following values:
  - X < 32
  - X = 32
  - X > 32

If it's too much we can split these in smaller tickets and I can tackle some when I finish the UI work.
cc @brandonfancher for visualization

Reply:
> awesome tests! Can you add a few more? (it does not need to be this pr)
> ...
>     * trying to induct with a pending member as inviter

This one is already there.